graphql_server/flask/graphqlview.py

```diff
@@ -84,18 +112,11 @@ def dispatch_request(self):
             pretty = self.pretty or show_graphiql or request.args.get("pretty")
 
             all_params: List[GraphQLParams]
-            execution_results, all_params = run_http_query(
-                self.schema,
-                request_method,
-                data,
-                query_data=request.args,
-                batch_enabled=self.batch,
-                catch=catch,
-                # Execute options
-                root_value=self.get_root_value(),
-                context_value=self.get_context(),
-                middleware=self.get_middleware(),
-            )
+            if self.enable_async:
+                execution_results, all_params = asyncio.run(self.resolve_results_async(request_method, data, catch))
```

Comment:
I don't like the idea of creating the event loop inside the execution step as not every user uses `asyncio` for that, other alternatives could be `trio` as example.

Reply:
Th correct approach should be the same as sanic and aiohttp integrations does:

```python
execution_results, all_params = run_http_query(
                self.schema,
                request_method,
                data,
                query_data=request.args,
                batch_enabled=self.batch,
                catch=catch,
                # Execute options
                run_sync=not self.enable_async,
                root_value=self.get_root_value(),
                context_value=self.get_context(),
                middleware=self.get_middleware(),
            )
            exec_res = (
                [
                    ex if ex is None or isinstance(ex, ExecutionResult) else await ex
                    for ex in execution_results
                ]
                if self.enable_async
                else execution_results
            )
```